src/js/Manager/ContextMenuManager.js

```diff
@@ -21,17 +22,24 @@ class ContextMenuManager {
     init() {
         if(!SystemService.hasContextMenu()) return;
         subscribe('suggestions:updated', (e) => {this._updateContextMenu(e.suggestions).catch(ErrorManager.catch);});
+        MessageService.listen('setting.set', (message) => {
```

Comment:
This will not work the way you think it does.
This will only receive the setting if you change it in the options.

Use the SettingsService as seen [here](https://github.com/marius-wieschollek/passwords-webextension/blob/master/src/js/Manager/AutofillManager.js#L43) to fetch the setting right away.